tensordict/tensorclass.py

```diff
@@ -1235,13 +1298,43 @@ def get_parent_locals(cls, localns=localns):
             localns=localns,
             # globalns=globals(),
         )
+        if tensor_only:
+
+            def is_tensor_or_optional_tensor(type_hint):
```

Comment:
I wonder if the localization of functions can also have a certain impact. Here, `is_tensor_or_optional_tensor` and `get_parent_locals` are redefined on every call to `_get_type_hints`. Maybe they should leave outside, but maybe it would only bring a marginal improvement.

Reply:
this is within the definition of your tensorclass, which you presumably run only once in your code
```
@tensorclass(tensor_only=True)  # <-- here
class X:
     ....
if __name__ == "__main__":
     x = X(...)  # <-- not here
     x.attr  # <-- not here
```